### ISSUE TITLE
PixelFlasher reports that there is not enough storage

### ISSUE THREAD

**Issue by adepssimius** [badabing2005/PixelFlasher#117]
`ERROR: Please check available disk space, you do not have safe levels of availabe storage to flash without risk.`

OS: OSX 12.5 Monterey
Arch: x86_64
<img width="701" alt="image" src="https://github.com/badabing2005/PixelFlasher/assets/5257052/a0914e93-8ad1-48e2-9a09-198cf80015ff">
<img width="731" alt="image" src="https://github.com/badabing2005/PixelFlasher/assets/5257052/1cd90529-7193-4178-809a-f2ef569fc0d2">

Steps attempted:
Clearing my disk space to greater than 5 Gb(sic) (was 2.6 GB initially). I am now at ~22 GB.
Disabling the check for enough disk space in pixelflasher settings.

Expected result:
I am able to successfully flash the phone.

Actual result:
PixelFlasher attempts to flash, but fails reporting `ERROR: Please check available disk space, you do not have safe levels of availabe storage to flash without risk.`(sic) in the log.

Detailed log:
```
Loading Device list ...
This could take a while, please be patient.


Getting package com.topjohnwu.magisk path on the device ...
Package Path is: /data/app/~~wr6lOwWxXB70Ya3plfErpg==/com.topjohnwu.magisk-g5wY_WnOKKm51e5n4fRfKg==/base.apk

Selected Device on 2023-11-15 12:25:56:
    Device ID:                       [redacted]
    Device Model:                    felix
    Device Active Slot:              a
    Device Mode:                     adb
    Has init_boot partition:         True
    Device Bootloader Version:       felix-14.0-10529422
    Device is Rooted:                False
    Device Build:                    UP1A.231005.007
    Device API Level:                34
    Device Architecture:             arm64-v8a
    Device Kernel Version:           5.10
    sys_oem_unlock_allowed:          0
    ro.boot.flash.locked:            1
    ro.boot.vbmeta.device_state:     locked
    vendor.boot.vbmeta.device_state: 
    ro.product.first_api_level:      32
    ro.boot.warranty_bit:            0
    ro.boot.veritymode:              enforcing
    ro.boot.verifiedbootstate:       green
    vendor.boot.verifiedbootstate:   
    ro.warranty_bit:                 0
    ro.secure:                       1
    ro.zygote:                       zygote64
    ro.vendor.product.cpu.abilist:   arm64-v8a
    ro.vendor.product.cpu.abilist32: 
    Magisk Manager Version:          26.4:26400
    Magisk Path:                     /data/app/~~wr6lOwWxXB70Ya3plfErpg==/com.topjohnwu.magisk-g5wY_WnOKKm51e5n4fRfKg==/base.apk
        Checked for Package:         com.topjohnwu.magisk

Load time: 2 seconds
Selected Boot:
    File:                  magisk_patched_26400_be935517_7069b867.img
    Path:                  /Users/[redacted]/Library/Application Support/PixelFlasher/boot_images4/be9355176b0b6687ba63a8f12f7ca14ce73ba02c/magisk_patched_26400_be935517_7069b867.img
    SHA1:                  7069b867ae0359c8a6ef07d9845c25dd289d1e1b
    Patched:               True
    Patched Method:        app
    Patch Source SHA1:     be9355176b0b6687ba63a8f12f7ca14ce73ba02c
    Patched With Magisk:   26.4:26400
    Patched on Device:     felix
    Stock Boot:            False
    Init Boot:             True
    Date:                  2023-11-15 12:22:32
    Firmware Fingerprint:  felix-up1a.231105.003
    Firmware:              /Users/[redacted]/Downloads/felix-up1a.231105.003-factory-604afb3f.zip
    Type:                  firmware



2023-11-15 12:26:00 User initiated Flash Pixel Phone.

2023-11-15 12:26:00 ERROR: Please check available disk space, you do not have safe levels of availabe storage to flash without risk.
Aborting ...


2023-11-15 12:26:00 INFO: Flashing was aborted.
This could be user initiated or a problem encountered during flashing.

2023-11-15 12:26:00 Getting device: [redacted] state  ...
Current device mode: adb
You might need to manually reboot your device.

Updating connected devices ...

Getting package com.topjohnwu.magisk path on the device ...
Package Path is: /data/app/~~wr6lOwWxXB70Ya3plfErpg==/com.topjohnwu.magisk-g5wY_WnOKKm51e5n4fRfKg==/base.apk

Selected Device on 2023-11-15 12:26:00:
    Device ID:                       [redacted]
    Device Model:                    felix
    Device Active Slot:              a
    Device Mode:                     adb
    Has init_boot partition:         True
    Device Bootloader Version:       felix-14.0-10529422
    Device is Rooted:                False
    Device Build:                    UP1A.231005.007
    Device API Level:                34
    Device Architecture:             arm64-v8a
    Device Kernel Version:           5.10
    sys_oem_unlock_allowed:          0
    ro.boot.flash.locked:            1
    ro.boot.vbmeta.device_state:     locked
    vendor.boot.vbmeta.device_state: 
    ro.product.first_api_level:      32
    ro.boot.warranty_bit:            0
    ro.boot.veritymode:              enforcing
    ro.boot.verifiedbootstate:       green
    vendor.boot.verifiedbootstate:   
    ro.warranty_bit:                 0
    ro.secure:                       1
    ro.zygote:                       zygote64
    ro.vendor.product.cpu.abilist:   arm64-v8a
    ro.vendor.product.cpu.abilist32: 
    Magisk Manager Version:          26.4:26400
    Magisk Path:                     /data/app/~~wr6lOwWxXB70Ya3plfErpg==/com.topjohnwu.magisk-g5wY_WnOKKm51e5n4fRfKg==/base.apk
        Checked for Package:         com.topjohnwu.magisk
```

**Comment by adepssimius** [comment]
Support file, as requested in other issues. On v6.0.1.1
[support_2023-11-15_12-41-17.zip](https://github.com/badabing2005/PixelFlasher/files/13367669/support_2023-11-15_12-41-17.zip)

**Comment by SharkWipf** [comment]
+1, also running into this same issue on Fedora 37 with the latest release.
I saw it checks /tmp/, which is tmpfs ("RAM disk") by default on Fedora, and this machine is fairly anemic so it only had 8GB total available.
But even after changing this (crudely) and restarting the application, getting the same error:
`2023-11-15 19:52:56 ERROR: Please check available disk space, you do not have safe levels of availabe storage to flash without risk.
Aborting ...`

Rolling back to the last version I used, 5.8.2.0, everything works as expected, so it's a recent regression.

**Comment by badabing2005** [comment]
Sorry folks, it looks like the flag to disable the check is not enforced.
As for the wrong reporting of the free storage, it's an issue in the psutil library, I already switched to shutil and the code is merged but not build yet (trying to address the flag issue as well)

**Comment by badabing2005** [comment]
See if 6.0.1.2 solves the issue now that I switched modules.
For the life of me I can't figure out why disabling it didn't work for you guys.
Instead of holding you folks longer, I released 6.0.1.2

[Here's the code](https://github.com/badabing2005/PixelFlasher/blob/4069db5857528a9479e7ea1e548e846a8ea06a0b/modules.py#L2644-L2649) that checks for the flag, and it does evaluate to False when the flag is not set.
If you spot something that I'm missing, please do let me know.
I'll keep on checking, but all my attempts to even reproduce the issue has failed.

**Comment by adepssimius** [comment]
Python is not my daily driver, but it seems a bit of an ambiguous logical test and I think the evaluation order would be (`config item` AND `free space check`) OR `other free space check`.

If `other free space check` is evaluated as true, I don't think the config item will apply.

**Comment by badabing2005** [comment]
🤦  So amateur mistake, and I looked at the code so many times and didn't see it.
I need a break.
Thank you

**Comment by adepssimius** [comment]
Yeah I've had a lot of those and I will continue to have them, doesn't matter how senior you get :). I pulled down the 6.0.1.3 release to test. It looks like the dmg for 6.0.1.3 is actually the 6.0.1.2 version. The sha hash matches, but when it's opened and installed everything says 6.0.1.2 and I'm prompted to upgrade to 6.0.1.3.

My testing steps and results (on what looks like 6.0.1.2, all with dry runs obviously): 
1. I reduced my disk space to less than 5 GB while the space check flag was enabled. _Result:_ **prevented from flashing**
2. I disabled the space check. _Result:_ **prevented from flashing**
3. I removed the files I had created to take up space so that I had more than 5 GB free. _Result:_ **allowed to flash**

In short, it looks like the space check is now functioning correctly in 6.0.1.2 on OSX Monterey, but that the ignore space check flag is being ignored. I think you know that already though.

**Comment by SharkWipf** [comment]
My findings on Fedora with 6.0.1.3 Ubuntu 22.04 (07fa7be...) are mostly the same, except the ignore checkbox does indeed correctly work for me now.
If I fill up my /tmp/ and uncheck the checkbox, it correctly continues even when there isn't enough space.
Change the setting, and it warns again, as you'd expect it to work.

So on my end, aside from the update check mentioned above, everything seems to be working correctly now.

**Comment by adepssimius** [comment]
My guess would be that `get_free_space(get_config_path())` might evaluate differently on linux than OSX, returning false on linux in all cases for `get_free_space(get_config_path()) < 5` and a valid value for OSX. That would leave `self.config.check_for_disk_space and get_free_space() < 5` as the only side of the OR that matters.

**Comment by SharkWipf** [comment]
> What is that? A RH based linux that is also a debian based linux? I thought those diverged 15-20 years ago.

Haha, no, I meant the Ubuntu 22.04 release of PixelFlasher (there is no Fedora release), Ubuntu 22.04 release works out of the box (if you get a newer adb manually).

> Weird, it shouldn't work unless Python behaves differently on Fedora.

Yeah I'm not sure why it works, I even tested it multiple times to make sure, and just tested it again, it definitely does work as it should though.
It also definitely did _not_ work on the previous version.

- Space checking setting enabled, application restarted (just in case):
> debug: Path: /tmp - Total: 7.29 GB, Used: 4.32 GB, Free: 2.97 GB
> 
> 2023-11-16 20:35:51 ERROR: Please check available disk space, you do not have safe levels of availabe storage to flash without risk.
Aborting ...

(Sidenote, typo, "availab**L**e")

- Nothing changed except created enough space: Prompts to flash normally.

- Space checking setting disabled, application restarted then confirmed it's still disabled, space-filler back in place: Prompts to flash normally.

- Setting still disabled, created enough space: Of course also prompts to flash normally.
